src/components/form.js

```diff
@@ -1,44 +1,64 @@
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+//  import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import axios from 'axios';
+import { addBook } from '../redux/books/books';
 
-const Form = ({ addBook }) => {
-  const [book, setBook] = useState({});
+const Form = () => {
+  const dipatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('Action');
 
-  const onChange = (e) => {
-    setBook({ ...book, [e.target.name]: [e.target.value] });
+  const createBook = (newBook) => {
+    axios
+      .post(
+        'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/KP5PI7hEkMIt6Rn76UzC/books',
+        newBook,
+      )
```

Comment:
[OPTIONAL]

- Same here. You can put the API call in a separate module and call it wherever you need by exporting them. It is very useful when you have to do many API calls in different files.